<launch>
    <arg name="robot_name" default="my_gen3" />
    <arg name="marker_id"    value="582"/>
    <arg name="marker_size" value="0.035"/>
    
    <include file="$(find gen3_move_it_config)/launch/planning_context.launch">
      <arg name="load_robot_description" value="true"/>
    </include>
    <include file="$(find gen3_move_it_config)/launch/move_group.launch">
      <arg name="allow_trajectory_execution" value="true"/>
      <arg name="fake_execution" value="false"/>
      <arg name="info" value="true"/>
      <arg name="debug" value="false"/>
    </include>

    <include file="$(find my_control)/launch/aruco_ros.launch">
        <arg name="robot_name"  value="$(arg robot_name)" />
        <arg name="camera_name" value="camera" />
        <arg name="marker_id"   value="$(arg marker_id)"/>
        <arg name="marker_size" value="$(arg marker_size)"/>
    </include>
    
    <include file="$(find easy_handeye)/launch/calibrate.launch">
        <arg name="eye_on_hand" value="true" />
        <arg name="freehand_robot_movement" value="false" />
        
        <arg name="move_group_namespace"    value="/$(arg robot_name)" /> 
        <arg name="move_group"              value="arm" />

        <arg name="robot_base_frame"        value="$(arg robot_name)/base_link" />
        <arg name="robot_effector_frame"    value="$(arg robot_name)/end_effector_link" /> 
        <arg name="tracking_base_frame"     value="$(arg robot_name)/camera_link" /> 
        <arg name="tracking_marker_frame"   value="aruco_marker_$(arg marker_id)" /> 
        
        <arg name="start_rviz" value="false" />
        <arg name="start_sampling_gui" value="true" />
    </include>

</launch>